services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/table/TableEntityTransactionManagerTest.java

```
@@ -0,0 +1,292 @@
+		// call under test.
+		TableEntityTransactionManager.validateRequest(request);
+	}
+
+	@Test
+	public void testUpdateTableWithTransaction() throws Exception {
+		// call under test.
+		TableUpdateTransactionResponse results = manager
+				.updateTableWithTransaction(progressCallback, userInfo, request);
+		assertEquals(response, results);
+		// user must have write
+		verify(tableManagerSupport).validateTableWriteAccess(userInfo, tableId);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test(expected = TableUnavailableException.class)
+	public void testUpdateTableWithTransactionTableUnavailableException()
+			throws Exception {
+
+		when(
+				tableManagerSupport.tryRunWithTableExclusiveLock(
+						any(ProgressCallback.class), anyString(), anyInt(),
+						any(ProgressingCallable.class))).thenThrow(
+				new TableUnavailableException(null));
+		// call under test.
+		manager.updateTableWithTransaction(progressCallback, userInfo, request);
+	}
+	
+	@SuppressWarnings("unchecked")
+	@Test(expected = LockUnavilableException.class)
+	public void testUpdateTableWithTransactionLockUnavilableException()
+			throws Exception {
+
+		when(
+				tableManagerSupport.tryRunWithTableExclusiveLock(
+						any(ProgressCallback.class), anyString(), anyInt(),
+						any(ProgressingCallable.class))).thenThrow(
+				new LockUnavilableException());
+		// call under test.
+		manager.updateTableWithTransaction(progressCallback, userInfo, request);
+	}
+	
+	@SuppressWarnings("unchecked")
+	@Test(expected = RecoverableMessageException.class)
+	public void testUpdateTableWithTransactionRecoverableMessageException()
+			throws Exception {
+
+		when(
+				tableManagerSupport.tryRunWithTableExclusiveLock(
+						any(ProgressCallback.class), anyString(), anyInt(),
+						any(ProgressingCallable.class))).thenThrow(
+				new RecoverableMessageException());
+		// call under test.
+		manager.updateTableWithTransaction(progressCallback, userInfo, request);
+	}
+	

```

Add test for: Any other exception should throw a runtime exception.